src/main/java/com/champions/carsharingservice/service/impl/CarServiceImpl.java

```diff
@@ -0,0 +1,73 @@
+package com.champions.carsharingservice.service.impl;
+
+import com.champions.carsharingservice.dto.car.CarDto;
+import com.champions.carsharingservice.dto.car.CreateCarRequestDto;
+import com.champions.carsharingservice.exception.EntityNotFoundException;
+import com.champions.carsharingservice.mapper.CarMapper;
+import com.champions.carsharingservice.model.Car;
+import com.champions.carsharingservice.repository.CarRepository;
+import com.champions.carsharingservice.service.CarService;
+import jakarta.transaction.Transactional;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CarServiceImpl implements CarService {
+    private final CarRepository carRepository;
+    private final CarMapper carMapper;
+
+    @Override
+    @Transactional
+    public CarDto save(CreateCarRequestDto request) {
+        Car car = carMapper.toEntity(request);
+        car.setType(Car.CarType.valueOf(request.type()));
+        return carMapper.toDto(carRepository.save(car));
+    }
+
+    @Override
+    @Transactional
+    public List<CarDto> getAll() {
+        return carRepository.findAll().stream()
+                .map(carMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    @Transactional
+    public CarDto getById(Long id) {
+        return carMapper.toDto(carById(id));
+    }
+
+    @Override
+    @Transactional
+    public CarDto update(Long id, CreateCarRequestDto request) {
+        Car car = carById(id);
+        updateValues(car, request);
+        return carMapper.toDto(carRepository.save(car));
+    }
+
+    @Override
+    @Transactional
+    public void delete(Long id) {
+        if (!carRepository.existsById(id)) {
+            throw new EntityNotFoundException("Can't find car by id: " + id);
+        }
+        carRepository.deleteById(id);
+    }
+
+    private Car carById(Long id) {
+        return carRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find car by id: " + id)
+        );
+    }
+
+    private void updateValues(Car car, CreateCarRequestDto request) {
+        car.setModel(request.model());
+        car.setBrand(request.brand());
+        car.setType(Car.CarType.valueOf(request.type()));
+        car.setInventory(request.amountAvailable());
+        car.setDailyFee(request.dailyFee());
+    }
```

Comment:
Better delete this mapper and add method to mapper: 
`void updateCar(CreateCarRequestDto dto, @MappingTarget Car car);`
and use it in update method in service